File: chapter 05/recipe 5.2/UsingInteropAssembly/UsingInteropAssemblyTest.cs
using System;
using Interop.DniSimpleComVB;

namespace UsingInteropAssemblyTest
{
	/// <summary>
	/// Reference an interop assembly built using tlbimp.exe
	/// </summary>
	class UsingInteropAssemblyTest
	{
		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main(string[] args)
		{
			DniSimpleComLib.IDniSimpleComObj comIObj 
				= new DniSimpleComLib.DniSimpleComObj();
			int result = comIObj.AddSomeNumbers(1, 2);
			Console.WriteLine("Call Com Object: {0}: {1}", 
				comIObj.GetType(), result);

			DniSimpleCom.Interop.IDniSimpleComObj comIObjNs
				= new DniSimpleCom.Interop.DniSimpleComObj();
			result = comIObjNs.AddSomeNumbers(1, 2);
			Console.WriteLine("Call Com Object: {0}: {1}", 
				comIObjNs.GetType(), result);

			//use a VB6 COM component
			DniSimpleComVBObj vbComObj = new DniSimpleComVBObj();
			result = vbComObj.AddSomeNumbers(1, 2);
			Console.WriteLine("Call VB6 Com Object: {0}", result);

     	    //wait for input
            Console.WriteLine("Press any key to exit");      
            Console.Read(); 	    
		}
	}
}

File: chapter 05/recipe 5.2/UsingInteropAssemblyVB/UsingInteropAssemblyTest.vb
Imports Interop.DniSimpleComVB

Module UsingInteropAssemblyTest

    Sub Main()
        'use a VB6 COM component
        Dim vbComObj As DniSimpleComVBObj = New DniSimpleComVBObj()
        Dim result As Integer = vbComObj.AddSomeNumbers(1, 2)
        Console.WriteLine("Call VB6 Com Object: {0}", result)

        'wait for input
        Console.WriteLine("Press any key to exit")
        Console.Read()
    End Sub

End Module
